### ISSUE TITLE
Handful of issues

### ISSUE THREAD

**Issue by thewebb** [rschurade/Ingnomia#57]
No longer accepts any keyboard commands including 'Escape' for the menu. This could be because I flick from monitor to monitor whilst 'working'.

Some of my workshops are just ignored. In 'The Ghoulish Fat' nobody wants to use my Carpenter. Not even the designated 'Carpenter'. In 'Misery Mule' the distillery won't be constructed at all possibly because the Carpenter is being ignored again - I've been focused on 'The Ghoulish Fat'.

Carpenter has in its build queue:
Error: Craftname $AlarmBell

[save.zip](https://github.com/rschurade/Ingnomia/files/5061302/save.zip)

Oh, and I'm addicted again. Over a thousand hours on gnomoria before I kicked the habit. You folks ruin lives.

**Comment by thewebb** [comment]
To confirm keyboard controls disappear after I've moved from one screen and back to another - may be possible to replicate by repeatedly ALT-TABing.

Also - noticed that some of my Craft To commands seem to get suspended. Is that because they run out of mats to use? I'd rather they just continued when more mats appear.

**Comment by rschurade** [comment]
Craft To commands suspend when the number of items exist in the world and continue when the number drops below the one you specified.

**Comment by thewebb** [comment]
Gotcha!

**Comment by rschurade** [comment]
For future reference you only need to zip and upload the latest save, i.e. the dir with the highest number

**Comment by thewebb** [comment]
Will do.

Also, the designation of Guard zones is currently non-functional. As is the allocation of uniforms.

**Comment by rschurade** [comment]
Guard zones yes, allocation of uniform should work

**Comment by rschurade** [comment]
Also please try to keep issues seperate, if you find something that doesn't work please open a new ticket. Some stuff is not implemented yet tho

**Comment by rschurade** [comment]
Ok I just loaded the Goulish Fat and the carpenter is used.

**Comment by thewebb** [comment]
> Craft To commands suspend when the number of items exist in the world and continue when the number drops below the one you specified. At least that's how they should behave.

Going back to this - I have 1.8k raw stone but my Stonecutter keeps suspending my blocks. Also on Ghoulish Fat.

**Comment by thewebb** [comment]
> Also please try to keep issues seperate, if you find something that doesn't work please open a new ticket. Some stuff is not implemented yet tho

Will do! Sorry.

**Comment by thewebb** [comment]
> Ok I just loaded the Goulish Fat and the carpenter is used.

Bizzare. It just sits there empty for me.

**Comment by rschurade** [comment]
When I load that save there is even one Gnome (Togglevolt) currently working at the carpenter.

**Comment by thewebb** [comment]
Right - let me reload that save. 

> > Craft To commands suspend when the number of items exist in the world and continue when the number drops below the one you specified. At least that's how they should behave.
> 
> Going back to this - I have 1.8k raw stone but my Stonecutter keeps suspending my blocks. Also on Ghoulish Fat. For a workaround specifically mention which raw stone you want them to use - then it works ok. It seems to be the 'any raw stone' that triggers it to suspended.

This is a misunderstanding on my part - when the craft to limit has been reached, then it appears to suspend the job.

**Comment by thewebb** [comment]
> When I load that save there is even one Gnome (Togglevolt) currently working at the carpenter.

Right. Lemme reload that save. Since then, despite have two carpenters, plenty of planks and thousands of wood the queue hasn't shifted.

Are you sure they aren't accepting the job and then rejecting it?

**Comment by rschurade** [comment]
No, they carry the planks to the workshop and build crates

**Comment by thewebb** [comment]
I've gone back to the old save in question and Togglevolt is just sitting around by a food crate and occasionally doing some hauling. He won't go near the carpenter. Hang on. Lemme video it for you.

**Comment by rschurade** [comment]
dont

**Comment by rschurade** [comment]
Also please remove your crude workbenches once you have better workshops

**Comment by thewebb** [comment]
Erm. Ok.

**Comment by rschurade** [comment]
And use linked stockpiles

**Comment by rschurade** [comment]
Anyway, Togglevolt of the Goulish Fat is happily carpenting.

**Comment by rschurade** [comment]
The game is #17 of Goulish Fat the last one in your zip and the day is 4th day of Autumn, 00:00, when you load it you will see Togglevolt currently working at the carpenter.

**Comment by thewebb** [comment]
That must have been an earlier save, Togglevolt is sat on a food crate in #17. All the crates are build already in #17.

BUT, I have figured out the bottleneck. It is waiting on a furnace to be made, there isn't one queued up at the stonecarver. Once you queue that up, it builds the furnace workshop is built and is able to make coal which releases the ability to make torches. I haven't dug into the other items in the carpenters to see what is missing, but ordering Craft To 1-3 of all components it unlocks the economy and my gnomes can get working again.

I am now becoming wise on linked stockpiles - thanks for the pointer!

**Comment by rschurade** [comment]
closing this now as it is too cluttered with several topics and most likely outdated.
